Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/worker/CWAWorkerFactory.kt

```diff
@@ -25,13 +25,29 @@ class CWAWorkerFactory @Inject constructor(
         workerClassName: String,
         workerParameters: WorkerParameters
     ): ListenableWorker? {
-        Timber.v("Looking up worker for %s", workerClassName)
-        val factory = factories.entries.find {
+        Timber.v("Checking in known worker factories for %s", workerClassName)
+        val ourWorkerFactories = factories.entries.find {
             Class.forName(workerClassName).isAssignableFrom(it.key)
         }?.value
 
-        requireNotNull(factory) { "Unknown worker: $workerClassName" }
-        Timber.v("Creating worker for %s with %s", workerClassName, workerParameters)
-        return factory.get().create(appContext, workerParameters)
+        return if (ourWorkerFactories != null) {
+            Timber.v("It's one of ours, creating worker for %s with %s", workerClassName, workerParameters)
+            ourWorkerFactories.get().create(appContext, workerParameters).also {
+                Timber.i("Our worker was created: %s", it)
+            }
+        } else {
+            Timber.w("Unknown worker class, trying direct instantiation on %s", workerClassName)
+            workerClassName.toNewWorkerInstance(appContext, workerParameters).also {
```

Comment:
I'm wondering if we should even try to create unknown workers? Why are we asked to create them in the first place?

Reply:
The WorkManager code may use a "wrapping" worker for our worker, to check constraints.

I.e. the wrapping worker would check that "is not low battery", and that worker would then decide whether to launch our worker.